src/src/DataStructs/SettingsStruct.cpp

```diff
@@ -540,21 +541,30 @@ bool SettingsStruct_tmpl<N_TASKS>::getSPI_pins(int8_t spi_gpios[3]) const {
   spi_gpios[0] = -1;
   spi_gpios[1] = -1;
   spi_gpios[2] = -1;
-  if (InitSPI > 0) {
+  if (isSPI_valid()) {
     # ifdef ESP32
-    switch (InitSPI) {
-      case 1:
+    const SPI_Options_e SPI_selection = static_cast<SPI_Options_e>(InitSPI);
+    switch (SPI_selection) {
+      case SPI_Options_e::Vspi:
       {
         spi_gpios[0] = 18; spi_gpios[1] = 19; spi_gpios[2] = 23;
         break;
       }
-      case 2:
+      case SPI_Options_e::Hspi:
       {
         spi_gpios[0] = 14; // HSPI_SCLK
         spi_gpios[1] = 12; // HSPI_MISO
         spi_gpios[2] = 13; // HSPI_MOSI
         break;
       }
+      case SPI_Options_e::UserDefined:
+      {
+        spi_gpios[0] = SPI_SCLK_pin;
+        spi_gpios[1] = SPI_MISO_pin;
+        spi_gpios[2] = SPI_MOSI_pin;
+        break;
+      }
+      case SPI_Options_e::None:
       default:
```

Comment:
No need for the default here, as you handled all the existing options.
So better to leave the default out. This way the compiler can help you when you forgot one.
This way you can even leave out the ESP32 specific enum values on ESP8266 builds and the compiler may help you to check if all build specific enum values are used or left out.

Reply:
Unfortunately, the ESP32 HSPI setting that matches the ESP8266 pins has value 2, where the ESP8266 SPI enabled setting is stored as 1, so it could be a very confusing `#ifdef` around those options, or we just keep the current enum and preprocessor conditions. I would pick that last option.

Reply:
Sure, if things get more confusing, then don't do it.
Splitting the enum to let the compiler help you may be a nice feature, not a goal.

So I will merge it now.